src/PlusServer/Commands/vtkPlusWinProbeCommand.cxx

```diff
@@ -0,0 +1,942 @@
+      metaData[parameterName] = std::make_pair(IANA_TYPE_US_ASCII, std::to_string(depth_value));
+    }
+    else if(igsioCommon::IsEqualInsensitive(parameterName, vtkPlusWinProbeVideoSource::SET_ALL_ARFI_FOCAL_DEPTHS))
+    {
+      std::istringstream ss(attribs["Value"]);
+      std::string val;
+      for(int i = 0; i < 6; i++)
+      {
+        ss >> val;
+        status = device->SetARFIFocalPointDepth(i, stof(val));
+        if(status != PLUS_SUCCESS)
+        {
+          error += "Error setting Focal Depth " + std::to_string(i) + " to " + val + "\n";
+          break;
+        }
+      }
+
+      std::string success = status == PLUS_SUCCESS ? "true" : "false";
+      std::string status_msg = status == PLUS_SUCCESS ? "SUCCESS" : "FAIL";
+      resultString += " Success=\"" + success + "\"";
+      metaData[parameterName] = std::make_pair(IANA_TYPE_US_ASCII, status_msg);
+    }
+    else if(igsioCommon::IsEqualInsensitive(parameterName, vtkPlusWinProbeVideoSource::SET_ARFI_FOCAL_DEPTHS))
+    {
+      int depth_index = stoi(attribs["Index"]);
+      float depth_value = stof(attribs["Value"]);
+      status = device->SetARFIFocalPointDepth(depth_index, depth_value);
+
+      std::string success = status == PLUS_SUCCESS ? "true" : "false";
+      std::string status_msg = status == PLUS_SUCCESS ? "SUCCESS" : "FAIL";
+      resultString += " Success=\"" + success + "\"";
+      metaData[parameterName] = std::make_pair(IANA_TYPE_US_ASCII, status_msg);
+    }
+    else if(igsioCommon::IsEqualInsensitive(parameterName, vtkPlusWinProbeVideoSource::GET_ARFI_FOCAL_DEPTHS))
+    {
+      int depth_index = stoi(attribs["Index"]);
+      float depth_value = device->GetARFIFocalPointDepth(depth_index);
+
+      status = PLUS_SUCCESS;
+      resultString += " Success=\"true\"";
+      resultString += " Value=\"" + std::to_string(depth_value) + "\"";
+      resultString += " Index=\"" + std::to_string(depth_index) + "\"";
+      metaData[parameterName] = std::make_pair(IANA_TYPE_US_ASCII, std::to_string(depth_value));
+    }
+    else if (igsioCommon::IsEqualInsensitive(parameterName, vtkPlusWinProbeVideoSource::GET_FIRST_GAIN_VALUE))
+    {
+      res = std::to_string(device->GetFirstGainValue());
+
+      status = PLUS_SUCCESS;
+      resultString += " Success=\"true\"";
+      resultString += " Value=\"" + res + "\"";
+      metaData[parameterName] = std::make_pair(IANA_TYPE_US_ASCII, res);
+    }
+    else if (igsioCommon::IsEqualInsensitive(parameterName, vtkPlusWinProbeVideoSource::SET_FIRST_GAIN_VALUE))
+    {
+      double gain_value = stod(attribs["Value"]);
+      status = device->SetFirstGainValue(gain_value);
+
+      std::string success = status == PLUS_SUCCESS ? "true" : "false";
+      std::string status_msg = status == PLUS_SUCCESS ? "SUCCESS" : "FAIL";
+      resultString += " Success=\"" + success + "\"";
+      metaData[parameterName] = std::make_pair(IANA_TYPE_US_ASCII, status_msg);
+    }
```

Comment:
A minor thing, but to prevent issues keeping the natural order of `Set` then `Get` for all command processing would be helpful. Here you switched it to `Get` then `Set`.

Reply:
updated